### PR TITLE
Fix/handle failed nft deposits

### LINKED ISSUE #126: Support withdrawals of conflicting NFT assets
With the current implementation, if a user deposits an NFT into the hyperchain that already exists on the hyperchain, the mint of that NFT will fail on the hyperchain. The user will then not have an option to withdraw that NFT from the L1 contract. 

A possible solution for this conflict is to issue a withdrawal on a failed mint. 
This withdrawal issuance could be done in the case that a stx/fungible token mint fails too.

### PR DESCRIPTION
**Pull request by jbencin**
### Description

If an NFT deposit into subnets fails, refund that NFT to user on L1 chain

### Applicable issues
- fixes #126 

### Additional info (benefits, drawbacks, caveats)

I wasn't sure whether to include the `StacksTransactionEvent` for the withdrawal in the same `StacksTransactionReceipt` as the failed deposit, or to create a new one. Both ways seemed to work, so I went with the former because it's simpler. I can change this if necessary

### Checklist
- [x] Test coverage for new or modified code paths
- [ ] Changelog is updated
- [ ] Required documentation changes (e.g., `docs/rpc/openapi.yaml` and `rpc-endpoints.md` for v2 endpoints, `event-dispatcher.md` for new events)
- [ ] New clarity functions have corresponding PR in `clarity-benchmarking` repo
- [ ] New integration test(s) added to `bitcoin-tests.yml`

### CODE REVIEW

**Review by obycode — COMMENTED** [review]
This looks great! Looking forward to seeing it working in some tests.

File: src/chainstate/stacks/db/blocks.rs [hunk 1]
```diff
@@ -4746,22 +4747,70 @@ impl StacksChainState {
...
-                        contract_analysis: None,
-                        execution_cost,
-                        microblock_header: None,
-                        tx_index: 0,
-                    }),
+                    Ok((value, _, mut events)) => {
+                        match &value {
+                            Value::Response(r) if r.committed == false => {
+                                // TODO: Figure out how this macro works and log some variables
+                                info!("DepositNft op returned error. Issue withdrawal tx");
```
Comment (kantai) [inline src/chainstate/stacks/db/blocks.rs:4754]: I think it's helpful for readability here to separate the response status check from the handling:

```suggestion
                        // check if the deposit failed, and if so, create an auto-withdraw event and add it to
                        // L1 operation's receipt.
                        let deposit_op_failed = if let Value::Response(r) = &value {
                              r.committed == false
                        };
                        if deposit_op_failed {
                             info!("DepositNft op returned error. Creating auto-withdrawal event."; "sender" => %sender, "id" => id, "asset-contract" => %subnet_contract_id);
                             
                             let nft_withdraw_event = ...;
                             events.push(nft_withdraw_event);
                        }
```

**Review by kantai — COMMENTED** [review]
This LGTM, just had a comment on some light refactoring.

File: src/chainstate/stacks/db/blocks.rs (hunk 1 above)
Comment (jbencin) [reply src/chainstate/stacks/db/blocks.rs:4754]: Agreed, I will make the change

Comment (jbencin) [reply src/chainstate/stacks/db/blocks.rs]: Done. Also factored out creating the `StacksTransactionEvent` into a separate function to reduce the size of `process_deposit_nft_ops()`

**Review by kantai — APPROVED** [review]
LGTM!

**Pull request merged**